app/connectors/DesConnector.scala

```diff
@@ -34,15 +34,14 @@ class DesConnector @Inject() (httpClient: HttpClient, desConfig: ApplicationConf
 
   private val logger: Logger = Logger(this.getClass)
 
-  private def header(correlationID: String): HeaderCarrier =
-    HeaderCarrier(extraHeaders =
-      Seq(
-        "Authorization" -> desConfig.authorization,
-        "Environment"   -> desConfig.environment,
-        "CorrelationId" -> correlationID
-      )
+  private def header(correlationID: String): Seq[(String, String)] =
+    Seq(
+      "Authorization" -> desConfig.authorization,
+      "Environment"   -> desConfig.environment,
+      "CorrelationId" -> correlationID
```

Comment:
Sorry, going to be annoying again, can you do the same as tamc:
def explicitHeaders(implicit hc: HeaderCarrier): List[(String, String)] =
    List(
      HeaderNames.authorisation -> urlHeaderAuthorization,
      HeaderNames.xRequestId    -> hc.requestId.fold("-")(_.value),
      HeaderNames.xSessionId    -> hc.sessionId.fold("-")(_.value),
      "Environment"             -> urlHeaderEnvironment,
      "CorrelationId"           -> UUID.randomUUID().toString)

https://github.com/hmrc/tamc/pull/56/files

For the correlationId, just use a randomUUID. You cannot trust the value from the user to be unique and in fact I know that DWP reuse the same id on failures...

Reply:
Headers added